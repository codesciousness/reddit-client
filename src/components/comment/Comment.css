.Comment {
    background: rgba(255, 255, 255, 0.7);
    border-radius: 20px;
    box-shadow: 0 5px 10px rgba(145, 145, 165, 0.6);
    margin: 2rem auto;
    padding: 1rem 2rem;
}

.Comment:hover {
    background: white;
}

.Comment > * {
    margin: 0.5rem 0;
}

.Comment__author {
    color: steelblue;
}