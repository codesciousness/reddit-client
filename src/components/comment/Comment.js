import React from 'react';
import './Comment.css';
import PostTime from '../postTime/PostTime';

const Comment = ({ comment }) => {
    return (
        <section id={comment.id} className="Comment">
            <p className="Comment__author">{comment.author} | <PostTime time={comment.created} /></p>
            <p className="Comment__body">{comment.body}</p>
            <p className="Comment__ups"><i className="fas fa-arrow-up"></i> {comment.ups} Ups</p>
        </section>
    );
}

export default Comment;