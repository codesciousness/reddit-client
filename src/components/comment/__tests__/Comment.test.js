import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import Comment from '../Comment';

const comment = {
    id: 'abcd1e',
    author: 'ScienceReader',
    created: '1 hour ago',
    body: 'Very informative article!',
    ups: 10
};

jest.mock('../../postTime/PostTime', () => () => (<span data-testid="PostTime">{comment.created}</span>));

describe('<Comment />', () => {
    test('Comment component is defined', () => {
        expect(<Comment />).toBeDefined();
    });
    test('matches the snapshot', () => {
        const component = create(<Comment comment={comment} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly with props', () => {
        render(<Comment comment={comment} />);
        expect(screen.getByText('Very informative article!')).toBeInTheDocument();
        expect(screen.getByText('10 Ups')).toBeInTheDocument();
    });
    test('renders a PostTime component', () => {
        render(<Comment comment={comment} />);
        expect(screen.getByTestId('PostTime')).toBeInTheDocument();
    });
});