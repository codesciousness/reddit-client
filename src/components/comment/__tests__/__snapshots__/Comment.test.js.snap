// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> matches the snapshot 1`] = `
<section
  className="Comment"
  id="abcd1e"
>
  <p
    className="Comment__author"
  >
    ScienceReader
     | 
    <span
      data-testid="PostTime"
    >
      1 hour ago
    </span>
  </p>
  <p
    className="Comment__body"
  >
    Very informative article!
  </p>
  <p
    className="Comment__ups"
  >
    <i
      className="fas fa-arrow-up"
    />
     
    10
     Ups
  </p>
</section>
`;
